### ISSUE TITLE
Breaking change on aqua

### ISSUE THREAD

**Issue by Its-Alex** [jdx/mise#3338]
**Describe the bug**

Since aqua repositories has been added, aqua has been used as default source but some package (as [kubectx](https://github.com/ahmetb/kubectx) in my case) does not works with aqua version.

**To Reproduce**
1. Execute `mise install kubectx@0.9.5 aqua:ahmetb/kubectx`
2. Use kubectx with multiple kubeconfig:
```sh
$ kubectx
error: kubeconfig error: failed to load: cannot determine kubeconfig path: multiple files in KUBECONFIG are currently not supported
```
3. Install asdf version `mise install kubectx@0.9.5 asdf:https://gitlab.com/wt0f/asdf-kubectx`
4. Use kubectx with multiple kubeconfig:
```
```sh
$ kubectx
xx
...
```

**Expected behavior**

Don't you think `asdf` should be the default source to prevent breaking change?

**Comment by jdx** [comment]
it works for me. Just use the asdf plugin.

**Comment by syhol** [comment]
I previously had kubectx installed so it defaults to the asdf plugin backend for me, however I can also recreate the issue when running with `mise exec`
```
Code/sandbox/20241203
❯ mise exec asdf:https://gitlab.com/wt0f/asdf-kubectx -- kubectx

Code/sandbox/20241203
❯ mise exec aqua:ahmetb/kubectx -- kubectx
error: kubeconfig error: failed to load: cannot determine kubeconfig path: multiple files in KUBECONFIG are currently not supported
```

I followed the thread and figgured out why this is happening.

The asdf version downloads the "Source code.tar.gz" and uses the bash script version, while the aqua version downloads the proper go compiled binaries for the correct architecture "kubectx_v0.9.5_darwin_arm64.tar.gz"

Reading their docs:

> Stable versions of kubectx and kubens are small bash scripts that you can find in this repository.
>
> Starting with v0.9.0, kubectx and kubens are now rewritten in Go. They should work the same way (and we'll keep the bash-based implementations around) but the new features will be added to the new Go programs. Please help us test this new Go implementation by downloading the binaries from the [Releases page →](https://github.com/ahmetb/kubectx/releases)

And [searching for the error](https://github.com/search?q=repo%3Aahmetb%2Fkubectx%20%22multiple%20files%20in%20KUBECONFIG%20are%20currently%20not%20supported%22&type=code) the in repo shows a result in the go code.

There is an issue in the repo with 46 (now 47 with me) upvotes https://github.com/ahmetb/kubectx/issues/211

**Comment by jdx** [comment]
So it seems we're just now using a newer version of the tool that doesn't have this feature

**Comment by syhol** [comment]
tldr; Yes.

Before version 0.9.0 https://github.com/ahmetb/kubectx only contained a bash implementation

From version 0.9.0 onwards https://github.com/ahmetb/kubectx contains 2 implementations that are released together and _**supposed**_ to be compatible:
- The original bash implementation
  - No longer actively developed
  - Is used by `asdf:https://gitlab.com/wt0f/asdf-kubectx` for all versions 0.1 - 0.9.5 and onwards
- The go rewrite implementation
  - Is actively developed
  - Is still missing some features from the bash implementation (like multiple files in KUBECONFIG)
  - Is used by `aqua:ahmetb/kubectx` for 0.9.0 - 0.9.5 and onwards
    - `aqua:ahmetb/kubectx` doesn't support <0.9.0

**Comment by Its-Alex** [comment]
@jdx 

> it works for me. Just use the asdf plugin. If you were using the asdf plugin before mise will continue to use that so it's not a breaking change.

Some people following our documentation to get all dependencies are getting aqua versions with the bug, so yes this create a bug.

**Comment by jdx** [comment]
it's not a breaking change. Once https://github.com/jdx/mise/issues/3339 you will be able to lock the backend into a lockfile.

**Comment by Its-Alex** [comment]
@jdx I think I'm missing some context, `asdf` will be deprecated?

**Comment by jdx** [comment]
that's unlikely, no

**Comment by Its-Alex** [comment]
@jdx In that case, why not keep it as the default source?

**Comment by jdx** [comment]
I've explained this in the docs in several places

**Comment by Its-Alex** [comment]
@jdx Sorry, I didn't find it, can you give me a link?

**Comment by jdx** [comment]
https://mise.jdx.dev/dev-tools/backends/asdf.html
https://mise.jdx.dev/dev-tools/backends/aqua.html
https://mise.jdx.dev/registry.html
